optuna/multi_objective/samplers/_motpe.py

```diff
@@ -0,0 +1,637 @@
+    return int(np.floor(0.1 * x))
+
+
+def _default_weights_above(x: int) -> np.ndarray:
+    return np.ones(x)
+
+
+@experimental("2.3.0")
+class MOTPEMultiObjectiveSampler(TPESampler, BaseMultiObjectiveSampler):
+    """Multi-objective sampler using the MOTPE algorithm.
+
+    This sampler is a multiobjective version of :class:`~optuna.samplers.TPESampler`.
+
+    For further information about MOTPE algorithm, please refer to the following paper:
+
+    - `Multiobjective tree-structured parzen estimator for computationally expensive optimization
+      problems <https://dl.acm.org/doi/abs/10.1145/3377930.3389817>`_
+
+    Args:
+        consider_prior:
+            Enhance the stability of Parzen estimator by imposing a Gaussian prior when
+            :obj:`True`. The prior is only effective if the sampling distribution is
+            either :class:`~optuna.distributions.UniformDistribution`,
+            :class:`~optuna.distributions.DiscreteUniformDistribution`,
+            :class:`~optuna.distributions.LogUniformDistribution`,
+            :class:`~optuna.distributions.IntUniformDistribution`,
+            or :class:`~optuna.distributions.IntLogUniformDistribution`.
+        prior_weight:
+            The weight of the prior. This argument is used in
+            :class:`~optuna.distributions.UniformDistribution`,
+            :class:`~optuna.distributions.DiscreteUniformDistribution`,
+            :class:`~optuna.distributions.LogUniformDistribution`,
+            :class:`~optuna.distributions.IntUniformDistribution`,
+            :class:`~optuna.distributions.IntLogUniformDistribution`, and
+            :class:`~optuna.distributions.CategoricalDistribution`.
+        consider_magic_clip:
+            Enable a heuristic to limit the smallest variances of Gaussians used in
+            the Parzen estimator.
+        consider_endpoints:
+            Take endpoints of domains into account when calculating variances of Gaussians
+            in Parzen estimator. See the original paper for details on the heuristics
+            to calculate the variances.
+        n_startup_trials:
+            The random sampling is used instead of the MOTPE algorithm until the given number
+            of trials finish in the same study. 11 * number of variables - 1 is recommended in the
+            original paper.
+        n_ehvi_candidates:
+            Number of candidate samples used to calculate the expected hypervolume improvement.
+        gamma:
+            A function that takes the number of finished trials and returns the number of trials to
+            form a density function for samples with low grains. See the original paper for more
+            details.
+        weights:
+            A function that takes the number of finished trials and returns a weight for them. As
+            default, weights are automatically calculated by the MOTPE's default strategy.
+        seed:
+            Seed for random number generator.
+
```

Comment:
IMO, the comment of "the improvement by the LHS initialization" should be placed here.
```suggestion
    .. note::
        Initialization with Latin hypercube sampling may improve optimization performance.
        However, the current implementation only supports initialization with random sampling.
```